### PR TITLE
Add skill to allow robot to speak

### PR DESCRIPTION
**Pull request by Tom0Brien**
This PR adds a skill which allows the robot to speak. The text-to-speech library used is a python based command-line tool https://github.com/MycroftAI/mimic3 which is not ideal as `system` call is needed, the quality of this library is really good though and runs locally.

### CODE REVIEW

**Comment by ysims** [comment]
I feel like audio testing shouldn't be in the behaviour testing module. The testing could be simplified with a config value in the Say module containing a test sentence and then ignoring it if it's empty otherwise emit the task with the string.

File: module/skill/Say/src/Say.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        on<Provide<SayTask>>().then([this](const SayTask& say, const RunInfo& info) {
+            // Only say text if it is a new task
+            if (info.run_reason == RunInfo::NEW_TASK) {
+                // Play the requested audio using python command-line tool mimic3 and aplay
+                system(std::string("mimic3 '" + say.text + "' | aplay").c_str());
+            }
+
+            // Nod head to indicate that the robot is "talking"
+            if (say.nod) {
+                emit<Task>(load_script<LimbsSequence>("NodYes.yaml"));
```
Comment (ysims) [inline module/skill/Say/src/Say.cpp:37]: Does this work?? `LimbsSequence` should only do something with servo commands for the arms and legs. This should be `HeadSequence`.

**Comment by Tom0Brien** [comment]
> I feel like audio testing shouldn't be in the behaviour testing module. The testing could be simplified with a config value in the Say module containing a test sentence and then ignoring it if it's empty otherwise emit the task with the string.

Hmmm, I like the tester module having all the available tasks for testing running combinations of providers simultaneously and having it within the module itself would require a role which is setup to use Say which currently doesn't exist.

**Comment by ysims** [comment]
> > I feel like audio testing shouldn't be in the behaviour testing module. The testing could be simplified with a config value in the Say module containing a test sentence and then ignoring it if it's empty otherwise emit the task with the string.
> 
> Hmmm, I like the tester module having all the available tasks for testing running combinations of providers simultaneously and having it within the module itself would require a role which is setup to use Say which currently doesn't exist.

I don't really see the point though, there's not really a use case where it's necessary and speaking doesn't conflict with other modules. I like the idea of another role because then you don't have to compile so much just to run speech.

File: module/skill/Say/src/Say.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                                         sanitized_text.end());
+                }
+                log<NUClear::DEBUG>("Saying: ", sanitized_text);
+                system(std::string("mimic3 '" + sanitized_text + "' --voice '" + cfg.voice + "' | aplay -D"
+                                   + cfg.device_name)
+                           .c_str());
+
+                // Nod head to indicate that the robot is "talking"
+                if (say.nod) {
+                    emit<Task>(load_script<HeadSequence>("Say.yaml"));
```
Comment (ysims) [inline module/skill/Say/src/Say.cpp:55]: shouldn't this just be the NodYes script?

File: module/skill/Say/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+Include this module whenever the robot needs to vocalize or acknowledge messages via speech-like actions.
+
+## Consumes
+
+- `message::skill::Say`: A Task requesting to vocalize a text string. Contains the text to be spoken and an optional nod gesture request.
+
+## Emits
+
+- `message::actuation::LimbsSequence`: A Task to instruct the robot's limbs to perform specific sequences, such as a nodding gesture.
```
Comment (ysims) [inline module/skill/Say/README.md:17]:
```suggestion
- `message::actuation::HeadSequence`: A Task to instruct the robot's head to perform specific sequences, such as a nodding gesture.
```

File: module/skill/Say/src/Say.cpp (hunk 2 above)
Comment (Tom0Brien) [reply module/skill/Say/src/Say.cpp:55]: This new script is essentially NodYes but with multiple nods

**Review by ysims — DISMISSED** [review]

**Comment by ysims** [comment]
You need to merge main and run format again

**Review by ysims — APPROVED** [review]

**Pull request merged**